Delphi 227 walkthrough for advice seekers

Leandro13Silva13 opened the pull request.

Leandro13Silva13 committed:
DELPHI-227 fix:
*Create change set to add new walk_through_enabled column to user's table

Leandro13Silva13 committed:
DELPHI-227 fix:
*Create change set to add new walk_through_enabled column to user's table

Leandro13Silva13 committed:
DELPHI-227 feat:
*Create change set to add new enableAdviceSeekerWalkThrough setting
*Rename the previous committed change set
*Add the new enableAdviceSeekerWalkThrough object to the application-settings.json
*Updated master-digi.xml to add/run the new updates

Leandro13Silva13 committed:
DELPHI-227 fix:
*update dockerImage.yml

Leandro13Silva13 committed:
DELPHI-227 fix:
*update dockerImage.yml

Leandro13Silva13 committed:
DELPHI-227 fix:
*update dockerImage.yml

Leandro13Silva13 committed:
DELPHI-227 fix:
*run mvn spotless:apply

Leandro13Silva13 committed:
DELPHI-227 fix:
*run mvn spotless:apply

Leandro13Silva13 committed:
DELPHI-227 fix:
*update dockerImage.yml

vi-sudo commented:
Task linked: [DELPHI-227 [DELPHI] Task: Walkthrough for Advice Seekers - dev](https://app.clickup.com/t/6656196/DELPHI-227)

patric-dosch-vi commented on src/main/java/com/vi/migrationtool/userservice/UserServiceAddWalkThroughColumnToUsersTable.java:
```diff
@@ -0,0 +1,42 @@
+  }
+
+  private void createWalkThroughColumn(JdbcTemplate jdbcTemplate) {
+    try {
```
Inserting a column into the database should be done in UserService via liquibase => 
/src/main/resources/db/changelog. 
This has no place in MigrationService. Or what am I overlooking?

Leandro13Silva13 replied:
You are 100% correct, long time ago, this changes where done here, but not anymore. That is my mistake, I will remove this and, correctly, use the liquibase.
Thank you very much for pointing this out :)

patric-dosch-vi commented on src/main/java/com/vi/migrationtool/mongodb/AddEnableAdviceSeekerWalkThroughSettingMigrationTask.java:
```diff
@@ -0,0 +1,61 @@
+  }
+
+  private void applyMigration(
+      ApplicationSettingService applicationSettingService,
```
this is stored in Mongo? I think sadly we do not have something liquibase-like for mongo? 
then this is fine.

Leandro13Silva13 replied:
Yes, we do not unfortunatly :(

Leandro13Silva13 committed:
DELPHI-227 fix:
*Removed the add of the new user's table column, as correctly pointed on the PR's comments

patric-dosch-vi reviewed (approved):
hey @Leandro13Silva13, this looks fine now. 
Could you please take care of a github squash merge when merging this PR? Thanks.

roman-vi commented on src/main/resources/migrations/changeset/0030_add_enable_AdviceSeekerWalkThrough_setting/0030_add_enable_AdviceSeekerWalkThrough_setting.xml:
```diff
@@ -0,0 +1,13 @@
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.6.xsd">
+
+  <changeSet author="tkuzynow" id="add_documentation_enabled_applevel_setting">
```
why the author is Tomas?

Leandro13Silva13 replied:
Forgot to change it to me, thank you for pointing it out. :)

roman-vi commented on src/main/resources/schemas/application-settings.json:
```diff
@@ -84,6 +84,21 @@
         "readOnly"
       ]
     },
+    "enableAdviceSeekerWalkThrough": {
```
as we don't refactor 'enableWalkthrough' (I guess the effort is high) we should at least comment it as //enableConsultantWalkthrough

roman-vi replied:
btw. why we call settings as actions e.g., enable vs. enabled? Setting itself is passive property which doesn't perform an action itself. I would rather expect a verb in method name, not in property. Just interested.

Leandro13Silva13 replied:
We must not comment it, because we need both the enableConsultantWalkthrough and the enableAdviceSeekerWalkThrough flags. Since one is responsible for the consultants and the new one is for the AS, so we can activate or deactivate the logic seperatly for eact type of users.

Leandro13Silva13 replied:
Regarding your second question I am not sure, this was done like this long before I joinned the project and I was told "this is how it was decided" when I asked

roman-vi commented on src/main/java/com/vi/migrationtool/mongodb/AddEnableAdviceSeekerWalkThroughSettingMigrationTask.java:
```diff
@@ -0,0 +1,61 @@
+
+  private boolean enableAdviceSeekerWalkThroughExists(
+      ApplicationSettingsEntity applicationSettingsEntity) {
+    return applicationSettingsEntity.getEnableAdviceSeekerWalkThrough() != null
```
```suggestion
return nonNull(applicationSettingsEntity.getEnableAdviceSeekerWalkThrough())
        && nonNull(applicationSettingsEntity.getEnableAdviceSeekerWalkThrough().getValue());

```

roman-vi reviewed (approved):
approved with questions and suggestion

Leandro13Silva13 committed:
DELPHI-227 fix:
*correct author

Leandro13Silva13 committed:
DELPHI-227 fix:
*change != null to nonNull
